app/controllers/vehicles_controller.rb

```diff
@@ -47,4 +49,12 @@ def destroy
   def vehicle_params
     params.require(:vehicle).permit(:brand, :name, :model, :year, :comments)
   end
+
+  def resolve_dealership
+    if current_user.dealership?
+      current_user.dealership
+    else
+      Dealership.last
```

Comment:
The rule is to assign the last dealership to a vehicle if the current user isn't dealership? This seems a bit odd....Also, this method could live in a different place than the controller. Remember that this isn't controller responsibility to have business rules.

Reply:
Absolutely, i'm going to move it out to a service object to keep business rules out of controllers.

About the behavior: it does seem odd, but I just kept the same as it was before. The feature addressed users who have dealerships associated to them but said nothing about admin users.

Reply:
fixed on #4